src/sentry/api/endpoints/group_hashes.py

```diff
@@ -32,15 +37,30 @@ def get(self, request, group):
         :auth: required
         """
 
-        queryset = GroupHash.objects.filter(
-            group=group.id,
+        data_fn = partial(
+            lambda *args, **kwargs: raw_query(*args, **kwargs)['data'],
+            start=datetime.utcfromtimestamp(0),  # will be clamped to project retention
+            end=datetime.utcnow(),  # will be clamped to project retention
+            aggregations=[
+                ('argMax(event_id, timestamp)', None, 'event_id'),
+                ('max', 'timestamp', 'latest_event_timestamp')
+            ],
```

Comment:
Probably need to test this change against real data, it may be faster to run `max((timestamp, event_id))` and unpack the result instead of two separate result columns (but would want actually test it rather than just running on intuition here.)

Reply:
Had a conversation with Ted and James on slack and the two approaches are about even. So going to leave it as is.

(adding a comment here for posterity)